typescript/cli/src/context/strategies/signer/MultiProtocolSignerManager.ts

```diff
@@ -221,4 +227,51 @@ export class MultiProtocolSignerManager {
     );
     return this.getSpecificSigner<SigningHyperlaneModuleClient>(chain);
   }
+
+  async getAddress(chain: ChainName): Promise<Address> {
+    const metadata = this.multiProvider.getChainMetadata(chain);
+
+    switch (metadata.protocol) {
+      case ProtocolType.Ethereum: {
+        const signer = this.getEVMSigner(chain);
+        return signer.getAddress();
+      }
+      case ProtocolType.CosmosNative: {
+        const signer = this.getCosmosNativeSigner(chain);
+        return signer.account.address;
+      }
+      default: {
+        throw new Error(`Protocol Type ${metadata.protocol} not supported`);
+      }
+    }
+  }
+
+  async getBalance(
+    isDryRun: boolean,
+    address: Address,
+    chain: ChainName,
+    denom?: string,
+  ): Promise<BigNumber> {
```

Comment:
can we have all these params grouped in an object instead? i think after 2+ params it gets quite messy calling a function